### DISCUSSION TITLE
Connect remote to multiple hubs

### DISCUSSION THREAD

**Discussion by Kalindor71** [pybricks/support#423] [Ideas]
While playing around with the new functionality for the remote, I wondered if it is possible to connect the remote to multiple hubs at the same time. If it isn't, this might be an welcome feature (at least to me).

I'm working on getting multiple trains running and being controlled by a single remote.
By using the right remote buttons to select the train and the left to set the speed of said train.

**Comment by dlech** [comment]
We're not sure if this is technically possible. If the official LEGO firmware does this, then we can probably do it to, but if it doesn't, then it the remote probably wasn't designed with this in mind.

An alternative that should be possible in the future is hub-to-hub communication, so it should be possible to connect one hub to another hub and the remote at the same time and relay messages through the central hub.

**Reply by Kalindor71** [reply]
The official Lego hardware does support this scenario, tried and tested it myself...

Hub-to-hub communication would be great great and more useful in my scenario, but since this still is far up on the roadmap I though maybe this would be easier to implement.

**Reply by dlech** [reply]
> The official Lego hardware does support this scenario, tried and tested it myself...

How does this work? Is everything connected to a central app on a smart device?

**Reply by Kalindor71** [reply]
Nope,

From a smart device you can connect up to 4 hubs. They can all be individualy addressed, so I recon it is done with an array of hubs (max 4).
But also a single remote can be connected to multiple hubs, By swapping a single motor from port A to port B you can control them independently. See https://www.youtube.com/watch?v=2Ug9c0eeWx0 from 1:50min for how to connect multiple trains on one remote).

**Reply by laurensvalk** [reply]
Nice find!

**Reply by BertLindeman** [reply]
Maybe this one is interesting too:
[lego powered up: how to connect multiple trains](https://www.youtube.com/watch?v=QhDdUFlq41k)

**Comment by laurensvalk** [comment]
This would be possible if we had a `disconnect()` method, or some other way to disconnect. Then you could toggle through the trains with one of the buttons. The remote light could indicate which one is currently connected.

This shouldn't be too hard to add.

> I'm working on getting multiple trains running and being controlled by a single remote.
> By using the right remote buttons to select the train and the left to set the speed of said train.

This scenario could work in that way. We do something like it [here](https://youtu.be/kN47P6mksj4?t=72).
